### PR TITLE
chroot: use user namespaces only when necessary, check if they're enabled on error

### PR DESCRIPTION
**Pull request by nalind**
When running with chroot isolation, only create a new user namespace when we have mappings to set.

If we get an error when trying to create a new user namespace, check `/proc/sys/user/max_user_namespaces` to verify that user namespaces are even available.

### CODE REVIEW

**Comment by nalind** [comment]
This should resolve #1061 and resolve #1068.

**Comment by bparees** [comment]
may also be the resolution to https://github.com/containers/buildah/issues/1058 ?

**Comment by nalind** [comment]
I think so, at least the bits where we run afoul of not being able to create user namespaces even in cases where we don't need to do any ID mapping.  The unit test that's failing in CI appears to be only reproducible when running under podman, still digging.

File: unshare/unshare.c [hunk 1]
```diff
@@ -41,6 +46,22 @@ void _buildah_unshare(void)
 	if ((flags & CLONE_NEWUSER) != 0) {
 		if (unshare(CLONE_NEWUSER) == -1) {
 			fprintf(stderr, "Error during unshare(CLONE_NEWUSER): %m\n");
+			fp = fopen(_max_user_namespaces, "r");
+			if (fp != NULL) {
+				memset(buf, 0, sizeof(buf));
+				n_read = fread(buf, 1, sizeof(buf) - 1, fp);
+				if (n_read != 0) {
```
Comment (TomSweeneyRedHat) [inline unshare/unshare.c]: Personal nit, would you mind changing to `n_read > 0`.  Personal scars from using != in the past for zero checking.

Comment (nalind) [reply unshare/unshare.c]: Sure.

**Comment by TomSweeneyRedHat** [comment]
@nalind looks like Red Hat CI isn't happy and it doesn't appear to be a flake.

**Comment by rhatdan** [comment]
LGTM

**Comment by giuseppe** [comment]
LGTM

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**